core/src/main/java/org/elasticsearch/cluster/routing/RoutingService.java

```diff
@@ -118,7 +118,7 @@ public void onFailure(String source, Exception e) {
         } catch (Exception e) {
             rerouting.set(false);
             ClusterState state = clusterService.state();
-            logger.warn((Supplier<?>) () -> new ParameterizedMessage("failed to reroute routing table, current state:\n{}", state.prettyPrint()), e);
+            logger.warn((Supplier<?>) () -> new ParameterizedMessage("failed to reroute routing table, current state:\n{}", state), e);
```

Comment:
I wonder if this should be guarded by an `isWarnEnabled` to avoid building the cluster state string if not.